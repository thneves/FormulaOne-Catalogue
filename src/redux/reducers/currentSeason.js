import { DISPLAY_SEASON } from '../actions';

const initialState = 0;

const currentSeason = (state = initialState, action) => {
  switch (action.type) {
    case DISPLAY_SEASON:
      return state + action.payload;
    default:
      return state;
  }
};

export default currentSeason;
