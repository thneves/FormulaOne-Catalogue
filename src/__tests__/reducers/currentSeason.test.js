import currentSeason, { initialState } from '../../redux/reducers/currentSeason';

describe('Current Season Reducer', () => {
  it('should return initial state', () => {
    expect(currentSeason(undefined, {})).toMatchSnapshot();
  });

  it('should handle the displayed season', () => {
    expect(currentSeason(initialState,
      {
        type: 'DISPLAY_SEASON',
      })).toMatchSnapshot();
  });
});
