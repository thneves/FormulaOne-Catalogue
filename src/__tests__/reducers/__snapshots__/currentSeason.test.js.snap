// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Season Reducer should handle the displayed season 1`] = `undefined`;

exports[`Current Season Reducer should return initial state 1`] = `0`;
